docs/modules/pipe.md

```diff
@@ -0,0 +1,131 @@
+# pipe Module
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2019-07-18 | [Terry Ellison](https://github.com/TerryE) | [Terry Ellison](https://github.com/TerryE) | [pipe.c](../../app/modules/pipe.c)|
+
+The pipe module provides RAM-efficient a means of passing character stream of records from one Lua
+task to another.
+
+## pipe.create()
+
+Create a pipe.
+
+#### Syntax
+`pobj = pipe.create()`
+
+#### Parameters
+None
+
+#### Returns
+A pipe resource.
+
+
+## pobj:read()
+
+Read a record from a pipe object.
+
+Note that the recommended method of reading from a pipe is to user a reader function as described below.
+
+#### Syntax
+`pobj:read([size/end_char])`
+
+#### Parameters
+- `size/end_char`
+	- If numeric then a string of `size` length will be returned from the pipe.
+	- If a string then this is a single character delimiter, followed by an optional "+" flag.  The delimiter is used as an end-of-record to split the character stream into separate records.  If the flag "+" is  specified then the delimiter is also returned at the end of the record, otherwise it is discarded.
+  - If omitted, then this defaults to `"\n+"`
+
+Note that if the last record in the pipe is missing a delimiter or is too short, then it is still returned, emptying the pipe.
```

Comment:
It would be helpful to know this condition to be able to recreate the original content. Especially if a delimiter without + was given. 
Maybe a second return value ```incomplete```.

Reply:
Gregor, I disagree here.  If you want the delim then use the '+' option.  If you don't, then you don't.  I want to keep this interface simple.  If you want to test for it's existence then use `+` and 
`if rec:sub(-1) == delim`

Reply:
Sorry the default is `"\n"`, not `"\n+"`.  I will update this on the next update.